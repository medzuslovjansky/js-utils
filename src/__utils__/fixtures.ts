type FixtureGetter = () => [string, string, string, string][];
type StringsGetter = () => string[];

export const adjectives: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/adjectives.json');
export const nouns_feminine: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/nouns-feminine.json');
export const nouns_masculine_animate: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/nouns-masculine-animate.json');
export const nouns_masculine: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/nouns-masculine.json');
export const nouns_misc: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/nouns-misc.json');
export const nouns_neuter: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/nouns-neuter.json');
export const verbs_imperfect: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/verbs-imperfect.json');
export const verbs_perfect: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/verbs-perfect.json');
export const verbs_misc: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/verbs-misc.json');
export const pronouns_demonstrative: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-demonstrative.json');
export const pronouns_indefinite: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-indefinite.json');
export const pronouns_interrogative: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-interrogative.json');
export const pronouns_personal: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-personal.json');
export const pronouns_possessive: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-possessive.json');
export const pronouns_reciprocal: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-reciprocal.json');
export const pronouns_reflexive: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-reflexive.json');
export const pronouns_relative: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/pronouns-relative.json');
export const other: FixtureGetter = () =>
  jest.requireActual('../__fixtures__/other.json');
export const word_forms: StringsGetter = () =>
  jest.requireActual('../__fixtures__/word-forms.json');
