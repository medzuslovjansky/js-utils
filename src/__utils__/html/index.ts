export * as adjective from './adjective';
export * as noun from './noun';
export * as verb from './verb';
