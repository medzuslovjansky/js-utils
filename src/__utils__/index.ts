export * from './toHTML';
export * from './findLemmaById';
export * as fixtures from './fixtures';
