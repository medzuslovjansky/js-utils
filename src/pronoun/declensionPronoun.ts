import { declensionAdjective } from '../adjective';
import { stripDiacritics } from '../common';
import transliterate from '../transliterate';
import {
  ALONE,
  FIRST_PERSON,
  POSSESSIVE,
  REFLEXIVE,
  SECOND_PERSON,
  THIRD_PERSON,
} from './lookups';

/** @deprecated */
export interface SteenPronounParadigm {
  type: string;
  columns?: string[];
  cases?: Record<string, any>;
  casesSingular?: Record<string, any>;
  casesPlural?: Record<string, any>;
}

export function declensionPronounFlat(rawWord: string, pronounType: string) {
  const paradigm = declensionPronoun(rawWord, pronounType);
  return paradigm ? getDeclensionPronounFlat(paradigm) : [];
}

function getDeclensionPronounFlat(result: SteenPronounParadigm): string[] {
  if (!result) {
    return [];
  }
  const forms: any = [];
  if (result.cases) {
    forms.push(...Object.values(result.cases));
  }
  if (result.casesSingular) {
    forms.push(...Object.values(result.casesSingular));
  }
  if (result.casesPlural) {
    forms.push(...Object.values(result.casesPlural));
  }
  const dirty: string[] = forms
    .flat()
    .join('/')
    .replace(/ /g, '')
    .split('/')
    .filter(Boolean)
    .reduce((acc: string[], item: string) => {
      if (item.indexOf('(') !== -1 || item.indexOf(')') !== -1) {
        acc.push(item.replace(/[()]/g, ''));
        acc.push(item.replace(/\(.*\)/, ''));
      } else {
        acc.push(item);
      }
      return acc;
    }, []);
  return Array.from(new Set(dirty));
}

export function declensionPronoun(
  rawWord: string,
  pronounType: string,
): SteenPronounParadigm | null {
  // now we don't know how to decline the phrases
  if (rawWord.includes(' ')) {
    return null;
  }

  const word = stripDiacritics(transliterate(rawWord, 'isv-Latn'));
  if (pronounType === 'personal' || pronounType === 'reflexive') {
    if (FIRST_PERSON.includes(word)) {
      return {
        type: 'noun',
        columns: ['singular', 'plural'],
        cases: {
          nom: ['ja', 'my'],
          acc: ['mene (mę)', 'nas'],
          gen: ['mene', 'nas'],
          loc: ['mně', 'nas'],
          dat: ['mně (mi)', 'nam'],
          ins: ['mnojų', 'nami'],
        },
      };
    } else if (SECOND_PERSON.includes(word)) {
      return {
        type: 'noun',
        columns: ['singular', 'plural'],
        cases: {
          nom: ['ty', 'vy'],
          acc: ['tebe (tę)', 'vas'],
          gen: ['tebe', 'vas'],
          loc: ['tobě', 'vas'],
          dat: ['tobě (ti)', 'vam'],
          ins: ['tobojų', 'vami'],
        },
      };
    } else if (THIRD_PERSON.includes(word)) {
      return {
        type: 'adjective',
        casesSingular: {
          nom: ['on', 'ono', 'ona'],
          acc: ['(n)jego (go)', '(n)jego (go)', '(n)jų'],
          gen: ['(n)jego', '(n)jej'],
          loc: ['(n)jem', '(n)jej'],
          dat: ['(n)jemu (mu)', '(n)jej'],
          ins: ['(n)jim', '(n)jejų'],
        },
        casesPlural: {
          nom: ['oni / one', 'one'],
          acc: ['(n)jih / (n)je', '(n)je'],
          gen: ['(n)jih'],
          loc: ['(n)jih'],
          dat: ['(n)jim'],
          ins: ['(n)jimi'],
        },
      };
    } else if (REFLEXIVE.includes(word)) {
      return {
        type: 'noun',
        columns: ['wordForm'],
        cases: {
          nom: [null],
          acc: ['sebę (sę)'],
          gen: ['sebe'],
          loc: ['sobě'],
          dat: ['sobě (si)'],
          ins: ['sobojų'],
        },
      };
    } else if (ALONE.includes(word)) {
      return {
        type: 'adjective',
        casesSingular: {
          nom: ['sam', 'sama', 'samo'],
          acc: ['samogo / sam', 'samų', 'samogo / samo'],
          gen: ['samogo', 'samoj', 'samogo'],
          loc: ['samom', 'samoj', 'samom'],
          dat: ['samomu', 'samoj', 'samomu'],
          ins: ['samym', 'samojų', 'samym'],
        },
        casesPlural: {
          nom: ['sami / same', 'same'],
          acc: ['samyh / same', 'same'],
          gen: ['samyh'],
          loc: ['samyh'],
          dat: ['samym'],
          ins: ['samymi'],
        },
      };
    }
  } else if (pronounType === 'possessive' && POSSESSIVE.includes(word)) {
    return {
      type: 'noun',
      columns: ['wordForm'],
      cases: {
        nom: [rawWord],
        acc: [rawWord],
        gen: [rawWord],
        loc: [rawWord],
        dat: [rawWord],
        ins: [rawWord],
      },
    };
  } else if (
    ['indefinite', 'interrogative', 'negative', 'relative'].includes(
      pronounType,
    ) &&
    (rawWord.includes('čto') ||
      rawWord.includes('kto') ||
      rawWord === 'kogo') &&
    !rawWord.includes('ktory')
  ) {
    let prefix = '';
    let postfix = '';
    let origWord = rawWord;
    if (rawWord === 'kogo') {
      origWord = 'kto';
    } else if (word.match(/koli$/)) {
      postfix = 'koli';
    } else if (word.match(/nebud$/)) {
      postfix = '-nebųď';
    } else if (word.match(/libo$/)) {
      postfix = '-libo';
    } else if (word.match(/^ino/)) {
      prefix = 'ino';
    } else if (word.match(/^ne/)) {
      prefix = 'ně';
    } else if (word.match(/^ni/)) {
      prefix = 'ni';
    } else if (word.match(/^vse/)) {
      prefix = 'vse';
    }
    if (origWord.match(/čto/)) {
      return {
        type: 'noun',
        columns: ['Word Form'],
        cases: {
          nom: [prefix + 'čto' + postfix],
          acc: [prefix + 'čego' + postfix],
          gen: [prefix + 'čego' + postfix],
          loc: [prefix + 'čem' + postfix],
          dat: [prefix + 'čemu' + postfix],
          ins: [prefix + 'čim' + postfix],
        },
      };
    } else if (origWord.match(/kto/)) {
      return {
        type: 'noun',
        columns: ['Word Form'],
        cases: {
          nom: [prefix + 'kto' + postfix],
          acc: [prefix + 'kogo' + postfix],
          gen: [prefix + 'kogo' + postfix],
          loc: [prefix + 'kom' + postfix],
          dat: [prefix + 'komu' + postfix],
          ins: [prefix + 'kym' + postfix],
        },
      };
    } else {
      return null;
    }
  } else if (pronounType === 'relative' && rawWord === 'iže') {
    return {
      type: 'adjective',
      casesSingular: {
        nom: ['iže', 'iže', 'iže'],
        acc: ['(n)jegože', '(n)jegože', '(n)jųže'],
        gen: ['(n)jegože', '(n)jejže'],
        loc: ['(n)jemže', '(n)jejže'],
        dat: ['(n)jemuže', '(n)jejže'],
        ins: ['(n)jimže', '(n)jejųže'],
      },
      casesPlural: {
        nom: ['iže', 'iže'],
        acc: ['(n)jihže / (n)ježe', '(n)ježe'],
        gen: ['(n)jihže'],
        loc: ['(n)jihže'],
        dat: ['(n)jimže'],
        ins: ['(n)jimiže'],
      },
    };
  } else if (
    [
      'demonstrative',
      'indefinite',
      'interrogative',
      'relative',
      'possessive',
      'negative',
      'universal',
    ].includes(pronounType)
  ) {
    const [origWord, postfix] = splitPostfix(rawWord, word);
    const adjectiveParadigm = declensionAdjective(origWord, postfix);
    return {
      type: 'adjective',
      casesSingular: adjectiveParadigm.singular,
      casesPlural: adjectiveParadigm.plural,
    };
  }
  return null;
}

const DEMONSTRATIVE_PRONOUNS: Record<string, string> = {
  ona: 'onȯj',
  ono: 'onȯj',
  onoj: 'onȯj',
  ota: 'otȯj',
  oto: 'otȯj',
  otoj: 'otȯj',
  ova: 'ov',
  ovo: 'ov',
  sa: 'sėj',
  se: 'sėj',
  sej: 'sėj',
  tamta: 'tamtȯj',
  tamto: 'tamtȯj',
  tamtoj: 'tamtȯj',
  tuta: 'tutȯj',
  tuto: 'tutȯj',
  tutoj: 'tutȯj',
  ta: 'tȯj',
  to: 'tȯj',
  toj: 'tȯj',
  te: 'tȯj',
  vsa: 'veś',
  vse: 'veś',
  vsi: 'veś',
};

const DEMONSTRATIVE_POSTFIXES: Record<string, string> = {
  ze: 'že',
  koli: 'koli',
  libo: '-libo',
  nebud: '-nebųď',
};

function splitPostfix(original: string, standard: string) {
  const [match = '', postfix = ''] =
    standard.match(/-?(nebud|koli|libo|ze)$/) || [];
  const main = original.slice(0, -match.length || undefined);

  return [
    DEMONSTRATIVE_PRONOUNS[main] || main,
    DEMONSTRATIVE_POSTFIXES[postfix] || postfix,
  ];
}
