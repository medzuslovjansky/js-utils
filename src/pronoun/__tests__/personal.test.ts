import { declensionPronoun } from '../declensionPronoun';
import { pronouns_personal } from '../../__utils__/fixtures';

describe('pronoun', () => {
  describe('personal', () => {
    test.each(pronouns_personal())('%s', (_id, _morphology, lemma) => {
      const actual = declensionPronoun(lemma, 'personal');
      expect(actual).toMatchSnapshot();
    });
  });
});
