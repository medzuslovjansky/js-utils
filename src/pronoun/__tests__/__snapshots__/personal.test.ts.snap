// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pronoun personal 73 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 675 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 716 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 792 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 982 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 1183 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 1507 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 1553 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 1602 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 1629 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 1816 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 1871 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 1935 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 2150 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 2219 1`] = `
{
  "casesPlural": {
    "acc": [
      "samyh / same",
      "same",
    ],
    "dat": [
      "samym",
    ],
    "gen": [
      "samyh",
    ],
    "ins": [
      "samymi",
    ],
    "loc": [
      "samyh",
    ],
    "nom": [
      "sami / same",
      "same",
    ],
  },
  "casesSingular": {
    "acc": [
      "samogo / sam",
      "samų",
      "samogo / samo",
    ],
    "dat": [
      "samomu",
      "samoj",
      "samomu",
    ],
    "gen": [
      "samogo",
      "samoj",
      "samogo",
    ],
    "ins": [
      "samym",
      "samojų",
      "samym",
    ],
    "loc": [
      "samom",
      "samoj",
      "samom",
    ],
    "nom": [
      "sam",
      "sama",
      "samo",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 2413 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 2603 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 2722 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 2771 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 2821 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 2925 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 3015 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 4803 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 4804 1`] = `
{
  "cases": {
    "acc": [
      "mene (mę)",
      "nas",
    ],
    "dat": [
      "mně (mi)",
      "nam",
    ],
    "gen": [
      "mene",
      "nas",
    ],
    "ins": [
      "mnojų",
      "nami",
    ],
    "loc": [
      "mně",
      "nas",
    ],
    "nom": [
      "ja",
      "my",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 4805 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 4809 1`] = `
{
  "cases": {
    "acc": [
      "tebe (tę)",
      "vas",
    ],
    "dat": [
      "tobě (ti)",
      "vam",
    ],
    "gen": [
      "tebe",
      "vas",
    ],
    "ins": [
      "tobojų",
      "vami",
    ],
    "loc": [
      "tobě",
      "vas",
    ],
    "nom": [
      "ty",
      "vy",
    ],
  },
  "columns": [
    "singular",
    "plural",
  ],
  "type": "noun",
}
`;

exports[`pronoun personal 4815 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 14576 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 14717 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;

exports[`pronoun personal 35621 1`] = `
{
  "casesPlural": {
    "acc": [
      "(n)jih / (n)je",
      "(n)je",
    ],
    "dat": [
      "(n)jim",
    ],
    "gen": [
      "(n)jih",
    ],
    "ins": [
      "(n)jimi",
    ],
    "loc": [
      "(n)jih",
    ],
    "nom": [
      "oni / one",
      "one",
    ],
  },
  "casesSingular": {
    "acc": [
      "(n)jego (go)",
      "(n)jego (go)",
      "(n)jų",
    ],
    "dat": [
      "(n)jemu (mu)",
      "(n)jej",
    ],
    "gen": [
      "(n)jego",
      "(n)jej",
    ],
    "ins": [
      "(n)jim",
      "(n)jejų",
    ],
    "loc": [
      "(n)jem",
      "(n)jej",
    ],
    "nom": [
      "on",
      "ono",
      "ona",
    ],
  },
  "type": "adjective",
}
`;
